azurerm/resource_arm_bot_channel_email.go

```diff
@@ -0,0 +1,198 @@
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Channel Email for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_bot_channel_email", *existing.ID)
+		}
+	}
+
+	channel := botservice.BotChannel{
+		Properties: botservice.EmailChannel{
+			Properties: &botservice.EmailChannelProperties{
+				EmailAddress: utils.String(d.Get("email_address").(string)),
+				Password:     utils.String(d.Get("email_password").(string)),
+				IsEnabled:    utils.Bool(true),
+			},
+			ChannelName: botservice.ChannelNameEmailChannel1,
+		},
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Kind:     botservice.KindBot,
+	}
+
+	if _, err := client.Create(ctx, resourceGroup, botName, botservice.ChannelNameEmailChannel, channel); err != nil {
+		return fmt.Errorf("Error issuing create request for Channel Email for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, botName, string(botservice.ChannelNameEmailChannel))
+	if err != nil {
+		return fmt.Errorf("Error making get request for Channel Email for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Channel Email for Bot %q (Resource Group %q): %+v", resourceGroup, botName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmBotChannelEmailRead(d, meta)
+}
+
+func resourceArmBotChannelEmailRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	botName := id.Path["botServices"]
+	resp, err := client.Get(ctx, id.ResourceGroup, botName, string(botservice.ChannelNameEmailChannel))
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Channel Email for Bot %q (Resource Group %q) was not found - removing from state!", id.ResourceGroup, botName)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Channel Email for Bot %q (Resource Group %q): %+v", id.ResourceGroup, botName, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", resp.Location)
+	d.Set("bot_name", botName)
+
+	if props := resp.Properties; props != nil {
+		if channel, ok := props.AsEmailChannel(); ok {
+			if channelProps := channel.Properties; channelProps != nil {
+				d.Set("email_address", channelProps.EmailAddress)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmBotChannelEmailUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).bot.ChannelClient
+	ctx := meta.(*ArmClient).StopContext
+
+	botName := d.Get("bot_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
```

Comment:
given this is the Update we should have the ID here - as such can we parse these from the ID?